src/Components/GEOSldas_GridComp/GEOSlandassim_GridComp/clsm_ensupd_read_obs.F90

```diff
@@ -1548,7 +1548,461 @@ subroutine read_obs_sm_ASCAT(                                  &
+    
+    !---------------------------------------------------------------------
+    ! 
+    ! Routine to read in ASCAT surface degree of saturation (sfds) obs. 
+    !
+    ! Read in EUMETSAT level 2 soil moisture product 25 km (SMO), PPF software version 5.0.
+    ! Data correspond to re-sampled (spatially averaged) backscatter (sigma0) values
+    ! on a 25-km orbit swath grid.  Input data files are in BUFR file format.
+    !
+    ! ASCAT_sm and ASCAT_sm_std outputs are in wetness fraction (i.e., 0-1) units!
+    !
+    ! Q. Liu,          Nov 2019 - based on read_obs_sm_ASCAT
+    ! A. Fox, reichle, Sep 2023 - updated
+    !
+    ! --------------------------------------------------------------------
+        
+    implicit none
+    
+    ! inputs:
+    
+    type(date_time_type), intent(in) :: date_time
+    
+    integer, intent(in) :: dtstep_assim, N_catd
+    
+    type(tile_coord_type), dimension(:), pointer :: tile_coord  ! input
+    
+    type(grid_def_type), intent(in) :: tile_grid_d
+    
+    integer, dimension(tile_grid_d%N_lon,tile_grid_d%N_lat), intent(in) :: &
+         N_tile_in_cell_ij
+    
+    integer, dimension(:,:,:), pointer :: tile_num_in_cell_ij   ! input
+    
+    type(obs_param_type), intent(in) :: this_obs_param
+    
+    ! outputs:
+   
+    logical, intent(out)                    :: found_obs
+
+    real,    intent(out), dimension(N_catd) :: ASCAT_sm              ! sfds obs          [fraction]  (i.e., 0-1)
+    real,    intent(out), dimension(N_catd) :: ASCAT_sm_std          ! sfds obs err std  [fraction]  (i.e., 0-1)
+    real,    intent(out), dimension(N_catd) :: ASCAT_lon, ASCAT_lat
+    real*8,  intent(out), dimension(N_catd) :: ASCAT_time            ! J2000 seconds
+    
+    ! ---------------
+    
+    ! Each obs file contains ~100-110 minutes (1 full orbit?) of observations (dt_ASCAT_obsfile).
+    ! File name indicates start time of swath.  
+    
+    integer,      parameter :: dt_ASCAT_obsfile = 110*60    ! seconds
+    
+    integer,      parameter :: N_fnames_max     = 15        ! max number of files per day
+
+    character(4), parameter :: J2000_epoch_id   = 'TT12'    ! see date_time_util.F90
+    
+    character( 15) :: str_date_time
+    character( 80) :: fname_of_fname_list
+    character(300) :: tmpfname, tmpfname2
+
+    type(date_time_type) :: date_time_tmp
+    type(date_time_type) :: date_time_low, date_time_low_fname
+    type(date_time_type) :: date_time_up
+    
+    integer :: ii, ind, N_tmp, N_files, kk, N_obs, N_fnames, N_fnames_tmp
+    
+    character(300), dimension(:), allocatable :: fnames, tmpfnames
+    
+    ! --------------------
+    !
+    ! variables for BUFR read
+
+    real*8, dimension(15) :: tmp_vdata
+
+    integer, parameter :: lnbufr  =     50       ! BUFR file unit number
+    integer, parameter :: max_rec =  20000       ! max number of obs after QC
+    integer, parameter :: max_obs = 250000       ! max number of obs read by subroutine
```

Comment:
I don't think there is a check if we exceed `max_obs`. If that ever happens, we probably get a nasty memory error.  Probably needs to be added.
A good value for `max_rec` depends on `dtstep_assim` and on whether we're reading just one orbit direction.  Here, too, we'll probably need a check.  Possibly we can make `max_rec` depend on `dtstep_assim` if we want to minimize memory demand, although that's probably not a priority now.

Reply:
add [error out](https://github.com/GEOS-ESM/GEOSldas/blob/3f95ee02780cf30259b9a2ea85eceb000822a7e7/src/Components/GEOSldas_GridComp/GEOSlandassim_GridComp/clsm_ensupd_read_obs.F90#L1786C1-L1789C23) if maxobs exceeded
```fortran
                if (N_obs > max_obs) then
                  err_msg = 'Attempting to read too many obs - how long is your assimilation window?'
                  call ldas_abort(LDAS_GENERIC_ERROR, Iam, err_msg)
                end if
```

Reply:
Still need to do the same with N_tmp and max_rec